source/MessageLogPanel.h

```diff
@@ -41,6 +43,7 @@ class MessageLogPanel : public Panel {
 private:
 	const std::deque<std::pair<std::string, Messages::Importance>> &messages;
 
+	const Interface *interf;
```

Comment:
Just call it `interface`. It should also be initialised to `nullptr`.

Though, I'm not sure the value of storing this here when it's only used in one place (other than immediately after it is initially retrieved).
It removes one string binary search every frame the panel is open with an empty log. Not a performance sensitive situation and probably the least common state of this panel.

In exchange for splitting the initialiser list i n the constructor across multiple lines, making it harder to read. Moving the initialisations into the constructor body would alleviate this, though.
It also makes it more difficult for a person reading the other place where it is used to determine what the name of the interface is. I'm sure other places do this, but those places probably also use their interface more than once.

Reply:
~~it was called `interface` but windows wasn't happy about it~~

At some point this pointer will make more sense, when we move more things to the interface, but I can remove it for now if you want.